SystemInformer/proctree.c

```diff
@@ -1451,6 +1486,155 @@ static VOID PhpUpdateProcessNodePriorityBoost(
+    _Inout_ PPH_PROCESS_NODE ProcessNode
+)
+{
+    if (!(ProcessNode->ValidMask & PHPN_SVCCOUNT))		// Bitwise flag for calling this update method
+    {
+        ProcessNode->SvcCount = 0;     // Default to nothing
+
+        PhAcquireQueuedLockShared(&ProcessNode->ProcessItem->ServiceListLock);
+
+        if ((ProcessNode->ProcessItem->ServiceList && ProcessNode->ProcessItem->ServiceList->Count != 0))
+        {
+
+            ULONG count = ProcessNode->ProcessItem->ServiceList->Count;
+            ProcessNode->SvcCount = count;									// Remeber this ULONG count for sorting on it
+            PhPrintInt32(ProcessNode->SvcCountText, count);					// What will be displayed in the column (converting ULONG to WCHAR array (PWSTR))
+        }
+
+        PhReleaseQueuedLockShared(&ProcessNode->ProcessItem->ServiceListLock);
+
+        ProcessNode->ValidMask |= PHPN_SVCCOUNT;
+    }
+}
+
+//
+//   This reads in the newly accessible list (ProcessNode->ProcessItem->ThreadList) from procprv.c that is always storing newly createdthread TIDs
+//
+static VOID PhpUpdateProcessNodeThreadTokenCount(_Inout_ PPH_PROCESS_NODE ProcessNode)
+{
+    if (!(ProcessNode->ValidMask & PHPN_THREADTOKENCOUNT))
+    {
+        ProcessNode->ThreadsWithTokenCounter = 0;
+
+        // I don't think this is neccessary here to check for ThreadListLock, as we are simply reading the list
+        //		it is not being modified
+
+        //PhAcquireQueuedLockShared(&ProcessNode->ProcessItem->ThreadListLock);
+
+        if (ProcessNode->ProcessItem->QueryHandle && ProcessNode->ProcessItem->ThreadList)
+        {
+            for (ULONG i = 0; i < ProcessNode->ProcessItem->ThreadList->Count; i++)
+            {
+                //SYSTEM_THREAD_INFORMATION* thread = ProcessNode->ProcessItem->ThreadList->Items[i];
+
+                // Instead, now just going to store TIDs instead of the entire SYSTEM_THREAD_INFORMATION
+                HANDLE* tid = ProcessNode->ProcessItem->ThreadList->Items[i];           // Basically the thread->ClientId.UniqueThread part of SYSTEM_THREAD_INFORMATION
+
+                //ULONG tidTest = HandleToUlong(thread->ClientId.UniqueThread);
+
+                // See thrdprv.c	:		for (i = 0; i < numberOfThreads; i++)
+
+                // ClientId.UniqueThread is a HANDLE, but its just a Thread ID. However leaving as Handle since PhOpenThread accepts it as one
+                HANDLE threadHandle = NULL;
+                if (NT_SUCCESS(PhOpenThread(&threadHandle, THREAD_QUERY_INFORMATION, tid)))
+                {
+                    // Success
+                }
+                else
+                {
+                    // Otherwise, try to open another way
+                    PhOpenThread(threadHandle, THREAD_QUERY_LIMITED_INFORMATION, tid);
+                }
+
+                if (threadHandle)
+                {
+                    // Like in thrdlist.c test to see if this thread has a token by attempting to open it
```

Comment:
I really think this should just live in the threads list for the process item and _not_ the processes list.

Reply:
The value is knowing which process has impersonating threads, otherwise you would literally have to click through hundreds of process to locate ones that have a thread token. Significance being: this process could be doing things that are far beyond its user or Process Token 

For example, a seemingly benign elevated user process could actually be impersonating (a thread within impersonating that is) say lsass (a process with the most available SePrivileges) with a System IL, have NT Service\TrustedInstaller present in its groups, and then perhaps even have SeCreateToken privilege to create arbitrary tokens (and hence processes) anywhere in any session.

Reply:
I do get this is out of scope of services names, but these were 5 very useful columns along with services names that I did AIO. But maybe I can break out each column in it’s PR-by-PR and discuss each in due time :-)

Reply:
Please open a _new_ enhancement issue on the topic of bubbling up the thread impersonation count to the process list.

Reply:
OK 👌🏼